### ISSUE TITLE
Development works, Production/Testflight crashes [iOS]

### ISSUE THREAD

**Issue by feruzm** [fechanique/cordova-plugin-fcm#168]
Any idea why app works on development and crashes when production/testflight built is tested.

I have no idea tracking on production, because in first run app asks permission for notification but crashes at the same time. after allowing, next starts quickly gets crashed without even splash screen showing up.

Any idea how to fix this? It is high likely that crash is caused by this plugin. Maybe there is small issue with start up getToken on iOS source code?

>iOS 10.0.2

**Comment by maxotov** [comment]
Someone is already using this plugin on production?

**Comment by wfdevelop** [comment]
Hi feruzm, I am facing the same problem for both ad-hoc and app-store provisioning on iOS. I just switched to production (appstore) and sent a build to Testflight, to see if it I get the same problem there. Unfortunately I do! I did not test in development (I guess ad-hoc is also kind of production and using the production aps-environment). 
Exactly the same behavior as described above. First asking permission for notification and then app crashes. But it is still in background! When I use home button to switch between apps it is still there, but cannot be chosen to run in foreground. It I select my app it is visible on screen for a short moment and then "crushing" again. 
Even when I remove all plugin calls in my JS code the same result.  The pure existence of the plugin makes the app crush.
Also using iOS 10.0.2.

**Comment by jcuervo** [comment]
@wfdevelop can you try adding (or modifying your existing line if you have it)
`<meta http-equiv="Content-Security-Policy" content="default-src * data: cdvfile: gap:; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' 'unsafe-eval'">`

we encountered that behaviour for all the features that require permissions (camera, push notifications, etc).

**Comment by wfdevelop** [comment]
@feruzm I could make it work in production and development environment finally. Before, it didn't run in any of both environments and the problem was in native iOS code! Finding this was really hard.

Finally I could find the error by plugging-in the iPhone with usb cable and checking the console output in Xcode at app start. There are hundreds of messages within seconds, so I deleted the log started my app and after the app crashed (after two seconds) I quickly stopped again by saving the log-output to disk. Then I went through every single line of output, until I found the reported error. In my case, the function for Firebase config was called a second time in native iOS code, which made the app crush at start. Perhaps it is not the same in your case but you can find the problem the same way!?

**Comment by Captnwalker1** [comment]
I had this same issue in AppDelegate+FCMPlugin.m there is [FIRApp configure]; (Line 50 for me) which the cordova-plugin-firebase already has a configure call in it's startup. Seems like this should be removed from the cordova-plugin-fcm

**Comment by jcuervo** [comment]
@Captnwalker1 i forked Gerjo's branch and added some stuff which disables the auto-connect feature. you can check it [https://github.com/jcuervo/cordova-plugin-fcm](url)

there are still other features that needs to be integrated in it (like the integration of the files from Firebase) and the token still gets fetched on startup but the permission can be controlled by the `connect` method.

**Comment by wfdevelop** [comment]
@Captnwalker1  Yes, I also simply set a comment on this line [FIRApp configure]. 
But I am not sure, if not to configure is right in every situation. I don't know exactly why FIRApp is already configured. Is it from JS-code or from another Firebase-Plugin I use!? 
I think it would be better (safer) to check if FIRApp is already configured and only if not yet, do [FIRApp configure] else do nothing. 
But I don't know exactly how to do in Object-C / Swift code, how to check previous configuration.

**Comment by Captnwalker1** [comment]
It comes from the firebase plugin dependency, https://github.com/arnesson/cordova-plugin-firebase, they have an AppDelegate+FirebasePlugin.m that has configure in didFinishLaunchingWithOptions as well. Didn't really dig but I didn't see anyway to see if firebase is configured already.

**Comment by siriocarino** [comment]
same problem fcm, in Development works in ios , but in production work only in android,
i use test https://cordova-plugin-fcm.appspot.com/ for send notification, any idea?

**Comment by mihairadulescu** [comment]
guys any update on this one ?

**Comment by leonardobazico** [comment]
Any updates about that?

**Comment by mihairadulescu** [comment]
i solved my issues by regenerating my certificates. no code change.

**Comment by SkateFreak** [comment]
@mihairadulescu can you please elaborate?
